bears/general/CasingBear.py

```diff
@@ -0,0 +1,153 @@
+        if "keywords" not in coalang or "special_chars" not in coalang:
+            self.warn("CasingBear did not run because 'keywords' and "
+                      "'special_chars' are necessary fields that are missing "
+                      "in the coalang definition for " + language)
+            return
+
+        self.keywords = coalang["keywords"]
+        self.delim = re.escape(str(coalang["special_chars"]) + " \t\n")
+
+        annotations = dependency_results[AnnotationBear.name][0].contents
+        string_results = annotations["strings"]
+        results = []
+        last_line = 1
+        last_column = 1
+        text = "".join(file)
+        for string_range in string_results:
+            results.append(SourceRange.from_values(
+                text,
+                start_line=last_line,
+                start_column=last_column,
+                end_line=string_range.start.line,
+                end_column=string_range.start.column))
+            last_line = string_range.end.line
+            last_column = string_range.end.column
+        results.append(SourceRange.from_values(
+            text,
+            start_line=last_line,
+            start_column=last_column,
+            end_line=len(file),
+            end_column=len(file[-1])))
+
+        changes = {}
+        for result in results:
+            line_number = result.start.line
+            while line_number <= result.end.line:
+                start = 0 if line_number != result.start.line \
+                    else result.start.column
+                end = result.end.column if line_number == result.end.line \
+                    else len(file[line_number - 1])
+
+                line_changes = self.process(file[line_number - 1][start:end])
+                for prev in line_changes:
+                    changes[prev] = line_changes[prev]
+
+                line_number += 1
+
+        lines_changed = set()
+        result_texts = []
+        for prev in changes:
+            change = changes[prev]
+            diff = Diff(file)
+            first_line = -1
+            num_changes = 0
+            skip_var = False
+
+            for line_number, line in enumerate(file, start=1):
+                rep = re.sub("(?P<g1>[\\" + self.delim + "]*)" + prev +
+                             "(?P<g2>[\\" + self.delim + "]*)",
+                             "\g<g1>" + change + "\g<g2>", line)
+
+                if rep != line:
+                    if line_number in lines_changed:
+                        skip_var = True
+                    lines_changed.add(line_number)
+                    diff.change_line(line_number, line, rep)
+                    num_changes += 1
+                    if first_line == -1:
+                        first_line = line_number
+
+            if skip_var:
+                continue
+
+            msg = "Change '" + prev + "' to '" + change + "'"
+            if num_changes > 1:
+                msg += ": " + str(num_changes) + " lines affected"
```

Comment:
uhm won't coala show this by default for large patches and show the patch for small ones so no need to do it in the message?